torch/distributions/normal.py

```diff
@@ -26,30 +27,25 @@ class Normal(Distribution):
 
     def __init__(self, mean, std):
         self.mean, self.std = broadcast_all(mean, std)
+        if isinstance(mean, Number) and isinstance(std, Number):
+            batch_shape = ()
+        else:
+            batch_shape = self.mean.size()
+        super(Normal, self).__init__(batch_shape)
 
     def sample(self, sample_shape=()):
-        if len(sample_shape) == 0:
-            return torch.normal(self.mean, self.std)
-        elif len(sample_shape) == 1:
-            return torch.normal(expand_n(self.mean, sample_shape[0]), expand_n(self.std, sample_shape[0]))
-        else:
-            raise NotImplementedError("sample is not implemented for len(sample_shape)>1")
+        shape = self._extended_shape(sample_shape)
+        return torch.normal(self.mean.expand(shape), self.std.expand(shape))
 
     def rsample(self, sample_shape=()):
-        if len(sample_shape) == 0:
-            eps = self.mean.new((self.mean + self.std).size())
-            eps.normal_()
-            return self.mean + self.std * eps
-        elif len(sample_shape) == 1:
-            expanded_mean = expand_n(self.mean, sample_shape[0])
-            expanded_std = expand_n(self.std, sample_shape[0])
-            eps = expanded_mean.new()
-            eps.normal_()
-            return expanded_mean + expanded_std * eps
-        else:
-            raise NotImplementedError("rsample is not implemented for len(sample_shape)>1")
+        shape = self._extended_shape(sample_shape)
```

Comment:
nit: This is already clear, but you could rely more on other ops to broadcast?
```py
shape = self._extended_shape(sample_shape)
eps = self.mean.new(shape).normal()
return self.mean + eps * self.std
```

Reply:
This is nicer. :) Will change.